docs/optional_features.md

```diff
@@ -0,0 +1,71 @@
+# Optional SSE features
+
+The core SSE specification was designed to support basic stream functionality in web applications, as defined by the `EventSource` API that is available in JavaScript in most browsers. However, for more complex use cases such as LaunchDarkly, it's desirable to have more sophisticated control over the SSE client. LaunchDarkly's SSE implementations therefore provide some additional features, whose behavior is described below.
+
+By default, the test harness will not run tests for these features, since an SSE implementation can be fully valid without including them. Tests for these are enabled based on the `capabilities` list returned by the test service. If the capability string for a particular features is in that list, then the test harness will run tests for it and will expect it to behave as described below.
+
+For more about the `capabilities` list, properties that the test harness can set in the client configuration, commands that the test harness may send, and callback messages that the test harness may expect, see [SSE test service specification](./service_spec.md).
+
+## Reading comments (capability: `"comments"`)
+
+This means that the SSE client can report comment lines to the caller.
+
```

Comment:
Added text in this section based on the conversation in https://github.com/launchdarkly/sse-contract-tests/pull/3.